src/ILLink.Shared/TrimAnalysis/HandleCallAction.cs

```diff
@@ -82,15 +86,18 @@ public bool Invoke (MethodProxy calledMethod, MultiValue instanceValue, IReadOnl
 				}
 
 				foreach (var value in instanceValue) {
-					if (value is SystemTypeValue typeValue)
-						AddReturnValue (new RuntimeTypeHandleValue (typeValue.RepresentedType));
-					else if (value is GenericParameterValue genericParameterValue)
-						AddReturnValue (new RuntimeTypeHandleForGenericParameterValue (genericParameterValue.GenericParameter));
-					else if (value == NullValue.Instance) {
-						// Throws if the input is null, so no return value.
-						returnValue ??= MultiValueLattice.Top;
-					} else
-						AddReturnValue (GetMethodReturnValue (calledMethod, returnValueDynamicallyAccessedMemberTypes));
+					if (value != NullValue.Instance)
```

Comment:
You're missing an else branch which would populate the return value to Top (the this worked before).

Reply:
Thanks for pointing that out, I hit some issues because of that.